addons/l10n_es_edi_tbai/models/xml_utils.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import hashlib
+import re
+from base64 import b64encode, encodebytes
+
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.asymmetric import padding
+from lxml import etree
+from odoo.tools.xml_utils import cleanup_xml_node
```

Comment:
Yes. I originally wanted to merge it first, but I got the feedback that it needs some real & merged use cases. So we're making and merging the use cases, then we'll dedup & add to tools. 
I tried to keep it as similar as possible with EC (and I think FacturaE is also almost the same) so that's why they overlap :)